src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,62 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+
+    private static final int SUPPLY_INDEX = 0;
+    private static final int BUY_INDEX = 1;
+    private static final int NAME_COLUMN = 0;
+    private static final int VALUE_COLUMN = 1;
+    private static final String SEPARATOR = ",";
+    private static final String BUY_WORD = "buy";
+    private static final String SUPPLY_WORD = "supply";
+
     public void getStatistic(String fromFileName, String toFileName) {
 
+        int[] data = parseAndAggregate(fromFileName);
+        makeReport(toFileName, data);
+    }
+
+    public int[] parseAndAggregate(String fileName) {
+        int[] results = new int[2];
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String value = reader.readLine();
+            String[] values;
+            while (value != null) {
+                values = value.split(SEPARATOR);
+                if (values[NAME_COLUMN].equals(SUPPLY_WORD)) {
+                    results[SUPPLY_INDEX] += Integer.parseInt(values[VALUE_COLUMN]);
+                }
+                if (values[NAME_COLUMN].equals(BUY_WORD)) {
+                    results[BUY_INDEX] += Integer.parseInt(values[VALUE_COLUMN]);
+                }
+                value = reader.readLine();
+            }
```

Comment:
It's bad practice to take a lot of logic in the try block. You should read data from the file and pass it to another method that calculates data and generate the report. Pass the report to the third private method that writes it to file (you can call method write only once)